copy/copy.go

```diff
@@ -313,16 +321,22 @@ func (ic *imageCopier) copyLayers() error {
 			if ic.diffIDsAreNeeded {
 				return errors.New("getting DiffID for foreign layers is unimplemented")
 			}
-			destInfo = srcLayer
+			destInfos[i] = destInfo
 			fmt.Fprintf(ic.reportWriter, "Skipping foreign layer %q copy to %s\n", destInfo.Digest, ic.dest.Reference().Transport().Name())
 		} else {
-			destInfo, diffID, err = ic.copyLayer(srcLayer)
-			if err != nil {
-				return err
-			}
+			eg.Go(func() error {
+				destInfo, diffID, err = ic.copyLayer(srcLayer)
+				if err != nil {
+					return err
+				}
+				destInfos[i] = destInfo
+				diffIDs[i] = diffID
```

Comment:
actually, `diffIDs` can contain less than `len(srcInfos)` - I'll shift it out of this loop to make sure it doesn't have holes (even if it's another loop, nothing which should impact performance though...I guess).